meson_options.txt

```diff
@@ -130,7 +130,7 @@ option('homed', type : 'feature', deprecated : { 'true' : 'enabled', 'false' : '
        description : 'install the systemd-homed stack')
 option('networkd', type : 'boolean',
        description : 'install the systemd-networkd stack')
-option('default-network', type : 'boolean', value : true,
+option('default-network', type : 'boolean', value : false,
```

Comment:
I'd leave it on, and just mention that this is switched in NEWS, and ask people to turn it of if they don't like it.

Reply:
mmh no this is just too disruptive, we'd need to manually prune it at the packaging level which is just silly, disabled by default is right

Reply:
why? you just do -Ddefault-network=disabled in your dpkg packaging scripts.

Packagers have to specify settings anyway, or is Debian just using vanilla settings.

Also, this is supposed to get distros to package these files in their own packages. We certainly won't get closer to this if noone notices it is there.

Reply:
Because in the vast, vast majority of cases you do _not_ want these to be enabled by default, it's the network management configuration stack that handles it and creates what's needed, according to what's configured. This is only a shortcut for local development and deployments, where you can just set the option in your local tree.

Reply:
I'm with Luca here, this is just going to cause more pain than gain. I'd leave it turned off by default

Reply:
No, absolutely not, these do things like enable dhcp by default on all interfaces that look like ethernet, it is the completely wrong place to set defaults like that - outside of local dev trees, which is fine to have the option, but disabled by default

Reply:
Then don't install the additional deb? or if you hate it so much, disable it via the meson option at debian package build time.

I mean, i don't use debian, so i don't care, but i am pretty sure that most people using networkd would probably like it if they just have to install `systemd-network-default.deb` or so and get networking to just work. 

And, I vehemently disagree with your idea that NM's behaviour of configuring networking by default was wrong. That's absolutely right. I mean, how ridiculous would that be if you don't get networking by default, but first have to enter some configuration dialog that asks you a bunch of questions noone can answer.

Reply:
WDYT about that 89-ethernet.network only matches with physical interface:
```
Kind=!*
```
c.f. c25aa6c8acc6d95eaacae7858a7057907d61a25e

Reply:
NM has a GUI with an indicator on the tray of what it's doing, _and_ an unprivileged user can disable it with a click. This is very, very much different from a configuration file hidden somewhere that requires root to change, and that by default starts blasting out of every interfaces, wired or wireless, and can't be stopped without admin intervention. It's not even close.

Reply:
@bluca that's simply not true on server systems.

And again: it's supposed to be packaged as separate RPM/DEB. So you only install it if you want automatic behaviour.

Reply:
> ```
> [Match]
> Kind=!*
> ```

yeah, that makes sense.

Reply:
There is no such separate package. If some distro wants to add it, then can also at the same time enable the meson config. What happens with the option enable by default is that EVERYONE starts shipping it in the default package, because that's how matching works. And that is just a non starter, because it can't be disabled without root privileges.

Reply:
"everyone"?

I dop hope packagers have a look at NEWS before they blindly take a new upstream release.

Reply:
and the goal with this change is to make distros *create* such a separate package.

Reply:
You might put too much faith in packagers. We have close contact with all the systemd maintainers of distributions we care about, so let's just work with them directly to make sure this is enabled and packaged separately. And for all the other distributions it'll be disabled by default until someone cares enough to enable it, but we don't really care about those.

Reply:
Yes, everyone, because such details are so small that they are going to be missed for sure. I had no idea it happened for example, fortunately the path was broken so while I shipped it by mistake it didn't do any damage.
And if anybody wants to create such a package, they can also enable the configuration at the same time, which is going to be trivial compared to the effort of adding a new package.

This is not an appropriate option to be enabled by default. It was added as a shortcut for developers machines, and on such machines it can be trivially be enabled in the build tree.

Reply:
I alreas said this should be announced in NEWS. And sorry, if a packager doesn't skim through NEWS, then they are shitty maintainers. And we might as well stop doing NEWS altogether if you think that not even packagers read that.

Reply:
Yeah, I am sorry, if people don't even read as far as "Announcements of Future Feature Removals and Incompatible Changes" then that's really sad, but I don't think we have to focus on those types of packages.

We should be willing to make changes of defaults wherever we need to, and downstreams can undo them, and so on, but I think communication via NEWS "incompat" section is the absolute bare minimum we can expect downstreams to read.

Reply:
> BTW, I asked it to be added, and I wanted it for a server machine and a VM. And I want that this is packaged, so that i can just list this in mkosi without having to sit down and write manual config.

Unrelated to this discussion but you can use `ip=dhcp` in the kernel command line and such which are parsed by systemd-network-generator to generate network units on the fly.

Reply:
After reading this discussion, now I'd like to give +1 for Lennart. Packagers (and who built systemd manually) should read NEWS file.

Reply:
People should do many things, but they don't - the question is, what happens when they don't? If a build breaks, or some new service is shipped, that's not too important in the grand scheme of things. But enabling by mistake all network and all wireless interfaces so that they start silently blasting packets whenever they are connected, without there being any way for an unprivileged user to stop it (unlike the much mentioned NM which has a clear GUI, indicator and toggle) is going to do a LOT of damage.

Reply:
I'm surprised that there's even discussion about this. Enabling this option by default creates a trap for users and packagers. It must be changed to be off by default.

> And sorry, if a packager doesn't skim through NEWS, then they are shitty maintainers.

Meh, by this definition, all maintainers I know are shitty maintainers. Sorry, but you have some 90's style idea of packaging in mind, where a maintainer has __the package__ and they carefully and thoughtfully approach each modification and then write a new `$Id$` header in every modified file. The reality is that packagers handle hundreds of packages each, and assume that upstream is not doing crazy things. I did more than skim the NEWS, I actually spent a few hours polishing them, but this doesn't mean that a week later I'm going to remember to add some unexpected optout when updating the package.

Reply:
Indeed. I even put together the NEWS, and yet completely failed to remember about this and shipped it in Debian - fortunately due to the path install bug it did no damage. I guess that makes me an extra shitty maintainer.

Reply:
I don't blame you, as this was hidden near the bottom of the NEWS file btw. As I said, this should be added to the top, where the compat changes are listed. But if noone looks there, as you say, then maybe we can drop that whole section too?

Reply:
This would be throwing out the baby with the bathwater. We always try to minimize breaking changes, and the general approach is to avoid them: we have 9–10 "breaking changes" listed, out of a total of 115 paragraphs in NEWS. The section in NEWS serves as a backup when compatibility needs to be broken. It's likely that some large fraction of people affected by those breaking changes will notice the section, and if they miss it and are unexpectedly affected, having it at the top makes it easy to find. But that doesn't change the fact we need to try to minimize incompatible changes in the first place.